pennylane/labs/resource_estimation/ops/qubit/non_parametric_ops.py

```diff
@@ -0,0 +1,612 @@
+
+    Args:
+        wires (Sequence[int] or int, optional): the wire the operation acts on
+
+    Resources:
+        The Y-gate can be decomposed according to the following identities:
+
+        .. math::
+
+            \begin{align}
+                \hat{Y} &= \hat{S} \cdot \hat{X} \cdot \hat{S}^{\dagger}, \\
+                \hat{X} &= \hat{H} \cdot \hat{Z} \cdot \hat{H}, \\
+                \hat{Z} &= \hat{S}^{2}, \\
+                \hat{S}^{\dagger} &= 3 \hat{S}.
+            \end{align}
+
+        Thus the resources for a Y-gate are six :class:`~.ResourceS` gates and
+        two :class:`~.ResourceHadamard` gates.
+
+    .. seealso:: :class:`~.Y`
+
+    **Example**
+
+    The resources for this operation are computed using:
+
+    >>> plre.ResourceY.resource_decomp()
+    [(6 x S), (2 x Hadamard)]
+    """
+
+    num_wires = 1
+
+    @classmethod
+    def default_resource_decomp(cls, **kwargs) -> list[GateCount]:
+        r"""Returns a list representing the resources of the operator. Each object represents a quantum gate
+        and the number of times it occurs in the decomposition.
+
+        Resources:
+            The Y-gate can be decomposed according to the following identities:
+
+            .. math::
+
+                \begin{align}
+                    \hat{Y} &= \hat{S} \cdot \hat{X} \cdot \hat{S}^{\dagger}, \\
+                    \hat{X} &= \hat{H} \cdot \hat{Z} \cdot \hat{H}, \\
+                    \hat{Z} &= \hat{S}^{2}, \\
+                    \hat{S}^{\dagger} &= 3 \hat{S}.
+                \end{align}
+
+            Thus the resources for a Y-gate are six :class:`~.ResourceS` gates and
+            two :class:`~.ResourceHadamard` gates.
+        """
+        s = resource_rep(ResourceS)
+        h = resource_rep(ResourceHadamard)
+
+        return [GateCount(s, 6), GateCount(h, 2)]
+
+    @property
+    def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: Empty dictionary. The resources of this operation don't depend on any additional parameters.
+        """
+        return {}
+
+    @classmethod
+    def resource_rep(cls) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute the resources."""
+        return CompressedResourceOp(cls, {})
+
+    @classmethod
+    def default_adjoint_resource_decomp(cls) -> list[GateCount]:
+        r"""Returns a list representing the resources for the adjoint of the operator.
+
+        Resources:
+            This operation is self-adjoint, so the resources of the adjoint operation results
+            in the original operation.
+
+        Returns:
+            list[GateCount]: A list of GateCount objects, where each object
+                represents a specific quantum gate and the number of times it appears
+                in the decomposition.
```

Comment:
All of these returns are not rendering well. I think it's due to the indent.
```suggestion
            list[GateCount]: A list of GateCount objects, where each object
            represents a specific quantum gate and the number of times it appears
            in the decomposition.
```